example/streaming/server.js

```diff
@@ -0,0 +1,169 @@
+            height:40px;
+        }
+        .skeleton.content {
+            height:60px;
+        }
+        .skeleton.footer {
+            height:60px;
+        }
+	`);
+});
+
+app.get(layout.pathname(), async (req, res) => {
+    const incoming = res.locals.podium;
+
+    incoming.view = {
+        title: 'Example streaming application',
+    };
+
+    const headerFetch = header.fetch(incoming);
+    const menuFetch = menu.fetch(incoming);
+    const contentFetch = content.fetch(incoming);
+    const footerFetch = footer.fetch(incoming);
+
+    incoming.hints.on('complete', async ({ js, css }) => {
+        // set the assets on httpincoming so that they are available in the document template
+        incoming.js = [...incoming.js, ...js];
+        incoming.css = [...incoming.css, ...css];
+
+        // set up the stream which will send the document template head
+        const stream = res.podiumStream();
+
+        // stream in the document body with slot placeholders for podlets
+        stream.send(`
+            <template shadowrootmode="open">
+                <link href="/foo/css" type="text/css" rel="stylesheet">
+                <div class="container">
+                    <div>
+                        <div>
+                            <slot name="header"><div class="skeleton header"></div></slot>
+                        </div>
+                    </div>
+                    <div>
+                        <div>
+                            <slot name="menu"><div class="skeleton menu"></div></slot>
+                        </div>
+                        <div>
+                            <slot name="content"><div class="skeleton content"></div></slot>
+                        </div>
+                    </div>
+                    <div>
+                        <div>
+                            <slot name="footer"><div class="skeleton footer"></div></slot>
+                        </div>
+                    </div>
+                </div>
+            </template>
+        `);
+
+        // fake 1 second delay
+        await new Promise((res) => setTimeout(res, 1000));
+
+        // stream in podlet content when available...
+        headerFetch.then((content) => {
+            stream.send(`<div slot="header">${content}</div>`);
+        });
+
+        await new Promise((res) => setTimeout(res, 1000));
+
+        menuFetch.then((content) => {
+            stream.send(`<div slot="menu">${content}</div>`);
+        });
+
+        await new Promise((res) => setTimeout(res, 1000));
+
+        contentFetch.then((content) => {
+            stream.send(`<div slot="content">${content}</div>`);
+        });
+
+        await new Promise((res) => setTimeout(res, 1000));
+
+        footerFetch.then((content) => {
+            stream.send(`<div slot="footer">${content}</div>`);
+        });
+
+        // close out the dom and the stream
+        await Promise.all([headerFetch, menuFetch, contentFetch, footerFetch]);
+        stream.done();
```

Comment:
Indicate that we are finished streaming content so that the html template can finish things off